miso-web/src/main/webapp/scripts/project_ajax.js

```diff
@@ -28,21 +28,22 @@ var projectId_sample, sampleQcTypesString, libraryQcTypesString, projectId_d3gra
 window.Parsley.addValidator('projectShortName', {
   validateString: function(value) {
     var deferred = new jQuery.Deferred();
-    Fluxion.doAjax('projectControllerHelperService', 'validateProjectShortName', {
-      'shortName': value,
-      'url': ajaxurl
-    }, {
-      'doOnSuccess': function(json) {
-        deferred.resolve();
-      },
-      'doOnError': function(json) {
-        deferred.reject(json.error);
-      }
+    jQuery.ajax({
+      url: '/miso/rest/project/validate-short-name',
+      type: 'POST',
+      contentType: 'application/json; charset=utf8',
+      data: JSON.stringify({
+        shortName: value
+      })
+    }).success(function(json) {
+      deferred.resolve();
+    }).fail(function(response, textStatus, serverStatus) {
+      deferred.reject(response); // need to upgrade Parsley to get custom error messages
     });
     return deferred.promise();
   },
   messages: {
-    en: 'Short name must conform to the naming scheme.'
+    en: 'Short name must be unique and must conform to the naming scheme.'
```

Comment:
The naming scheme dictates whether duplicates are allowed, so that should not be stated here